docs/user-guide/bandwidth-resources.md

```diff
@@ -0,0 +1,170 @@
+[here](http://releases.k8s.io/release-1.0/docs/user-guide/bandwidth-resources.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Bandwidth resources
+
+When specifying a [pod](pods.md), you can optionally specify how much bandwidth (bits/sec) each
+container can consume.  There are separate constraints for ingress and egress bandwidth. When pods
+have resource limits, the scheduler is able to make better
+decisions about which nodes to place pods on, and contention for resources can be handled in a
+consistent manner.
+
+*ingress* and *egress* bandwidth are each a *resource type*.  All resource types have a base unit.
+For both ingress and egress this unit is bits per second.
+
+Ingress and egress are collectively referred to as *bandwidth resources*.  Bandwidth
+resources are measureable quantities which can be requested, allocated, and consumed.  They are
+distinct from [API resources](working-with-resources.md).  API resources, such as pods and
+[services](services.md) are objects that can be written to and retrieved from the Kubernetes API
+server.
+
+## Pod resource limits
+
+Each Pod can optionally specify `spec.containers[].resources.limits.ingress` and/or
+`spec.containers[].resources.limits.egress`.
+
+Specifying resource limits is optional.  In some clusters, an unset value may be replaced with a
+default value when a pod is created or updated.  The default value depends on how the cluster is
+configured.
+
+Although limits can only be specified on individual containers, it is convenient to talk about pod
+resource limits.  A *pod resource limit* for a particular resource type is the sum of the resource
+limits of that type for each container in the pod, with unset values treated as zero.
+
+The following pod has two containers.  Each has a limit of 1Mbit of ingress bandwidth and 2Mbit of
+egress bandwidth.
+The pod can be said to have a limit of 2Mbit of ingress bandwidth and 4Mbit of egress bandwidth.
+
+*Note* Because of limitations in container runtimes, we can't typically limit the bandwidth between
+containers, so different containers in a Pod may steal from each other.  We can (and do) limit
+the overall bandwidth for the Pod.  We hope to refine this as capabilities for hierarchical
+network namespaces come into the container runtimes like Docker.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: frontend
+spec:
+  containers:
+  - name: db
+    image: mysql
+    resources:
+      limits:
+        ingress: "1M"
+        egress: "2M"
+  - name: wp
+    image: wordpress
+    resources:
+      limits:
+        ingress: "1M"
+        egress: "2M"
+```
+
+## How pods with resource limits are scheduled
+
+When a pod is created, the Kubernetes scheduler selects a node for the pod to
+run on.  Each node has a maximum capacity for each of the resource types: the
```

Comment:
How does the node report bandwidth capacity?  This alludes to CPU and memory, but does not discuss how ingress, and egress are expressed.  Does it require the operator to define this?